safedelete/managers.py

```diff
@@ -100,6 +100,27 @@ def all(self, **kwargs):
             qs._safedelete_force_visibility = force_visibility
         return qs
 
+    def update_or_create(self, defaults=None, **kwargs):
+        """
+        Regular update_or_create() fails on soft-deleted, existing record with unique constraint on non-id field
+        :param defaults:
+        :param kwargs:
+        :return:
+        """
```

Comment:
Please use Google Type docstrings. (That's what I gather is being used here primarily. @Gagaro  will know more.)